docs/okeps/okep-5494-ovn-kubernetes-mcp-server.md

```diff
@@ -0,0 +1,740 @@
+# OKEP-5494: ModelContextProtocol for Troubleshooting OVN-Kubernetes
+
+# Problem Statement
+
+Diagnosing an issue in [OVN-Kubernetes](https://ovn-kubernetes.io/) network
+plugin is complex because it has many layers (Kubernetes, OVN-Kubernetes, OVN,
+OpenvSwitch, Kernel - especially Netfilter elements). Usually the person
+troubleshooting an issue has to approach it in a layered fashion and has
+to be fully aware of all the debugging tools each layer has to offer to
+then be able to pin point where the problem is. That is time consuming
+and requires years of expertise on each depth of the stack and across all
+features. Sometimes it also involves working with the layered community
+project teams like OVN and OVS since they hold more knowledge about their
+domain than engineers working on the OVN-Kubernetes plugin. Each
+troubleshooting session to understand where the packet is getting
+backholed or dropped takes a lot of time to solve (unless it's trivial).
+
+# Goals
+
+The goal of this enhancement is to try to improve "troubleshooting time",
+"tool usage/typing time" and "erase the need to know how to use each of those
+tools to a parameter level detail" by exposing these tools using
+[Model Context Protocol](https://modelcontextprotocol.io/docs/learn/architecture#overview)
+and leveraging the backend Model's (Claude Sonnet4, Gemini2.5Pro, GPT-5, etc)
+knowledge and context to troubleshoot issues on live clusters or locally on
+containers with customer escalation's databases loaded. This can go a long
+way in speeding up bug triages.
+
+* Phase1 targets only ovn-kubernetes community members as target audience
+* Build an OVN-Kubernetes MCP Server(s) that exposes all the tools required
+  to troubleshoot OVN-Kubernetes network plugin
+    * This MCP Server(s) must also be aware of where to run these tools to
+      troubleshoot the issue in question on a cluster (i.e Which node? Which
+      pod? Which container?)
+* Add support to load a MCP Server against not just a real-time cluster
+  but also against a simulated environment constructed from a must-gather
```

Comment:
I believe that both `must-gather` and `sos report` are OpenShift specific, can we generalize this statement?
We could provide these as examples but please provide a link to the definition.

Reply:
done